haystack/preview/embedding_backends/sentence_transformers_backend.py

```diff
@@ -0,0 +1,50 @@
+from typing import List, Optional, Union, Dict
+import hashlib
+import numpy as np
+
+from haystack.preview.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as sentence_transformers_import:
+    from sentence_transformers import SentenceTransformer
+
+
+class SentenceTransformersEmbeddingBackendFactory:
+    """
+    Factory class to create instances of Sentence Transformers embedding backends.
+    """
+
+    _instances: Dict[str, "_SentenceTransformersEmbeddingBackend"] = {}
+
+    @staticmethod
+    def get_embedding_backend(
+        model_name_or_path: str, device: Optional[str] = None, use_auth_token: Union[bool, str, None] = None
+    ):
```

Comment:
@masci suggested that this method might allow a `fresh_istance` boolean parameter (default value: `False`)...

I'm unsure about how to handle it.
If an embedding backend is created with
`SentenceTransformersEmbeddingBackendFactory.get_embedding_backend(..., fresh_istance=True), how can I compute its id? Should we make it reusable?

Reply:
Don't do that - it was an example to advocate for a more future-proof design.

If and when this becomes a feature request, I think we can just give back a fresh instance and forget. The parameter could be called `pooling=False` and that's it.

Reply:
See for example how psycopg deals with a similar problem with connection pools https://www.psycopg.org/psycopg3/docs/advanced/pool.html#null-connection-pools

Reply:
Thanks for the clarification
I would prefer naming it `force_fresh_instance` or something similar,
because [pooling has a different meaning related to embeddings](https://blog.ml6.eu/the-art-of-pooling-embeddings-c56575114cf8).